ports/esp32/mphalport.h

```diff
@@ -136,6 +137,15 @@ static inline int mp_hal_pin_read(mp_hal_pin_obj_t pin) {
 static inline void mp_hal_pin_write(mp_hal_pin_obj_t pin, int v) {
     gpio_set_level(pin, v);
 }
+static inline int mp_hal_pin_read_output(mp_hal_pin_obj_t pin) {
+    #if defined(GPIO_OUT1_REG)
+    return pin < 32
+        ? (*(uint32_t *)GPIO_OUT_REG >> pin) & 1
+        : (*(uint32_t *)GPIO_OUT1_REG >> (pin - 32)) & 1;
+    #else
+    return *(uint32_t *)GPIO_OUT_REG >> pin & 1;
```

Comment:
Please put parenthesis around the `>>` operator and its operands, just to be certain the calculation is done with the correct precedence.

Reply:
Changed an just recompiled to be sure it builds.